src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2124,7 +2118,7 @@ int journal_file_append_entry(
 
         assert(f);
         assert(f->header);
-        assert(iovec || n_iovec == 0);
+        assert(iovec && n_iovec > 0);
```

Comment:
as above

Reply:
Both call sites of `journal_file_append_entry` have the same `assert` which is why I duplicated it here as well. Should those be changed to remove the `assert` as well?